### DISCUSSION TITLE
port_handler=slirp4netns not working?

### DISCUSSION THREAD

**Discussion by FGasper** [containers/podman#14422] [General]
Hello,

I’m running this:
```
podman run -it --network slirp4netns:port_handler=slirp4netns --publish 10.2.65.75:8080:80 almalinux:8 /bin/bash -c 'dnf install -qy iproute; ip route get 1'
```

I’m expecting to see `10.2.65.75` printed since I’m using `port_handler=slirp4netns`, but instead I see `10.0.2.100`.

Am I “holding it wrong”, or just misunderstanding what that option actually does? And, is there some way I could configure my container so that it _does_ show 10.2.65.75 (assuming that the container runs from a machine that uses that same IP)?

Thank you!

**Comment by Luap99** [comment]
I have no idea about your perl command but I guess you are looking at the local ip (which is 10.0.2.100 by default) and not the remote ip.

**Reply by FGasper** [reply]
Hi!

That’s right: I need the container’s local IP to be 10.2.65.75, not 10.0.2.100. Is there some way I can get there?

(Updated the original post with a simpler demonstration.)

**Reply by Luap99** [reply]
This already works you are just looking at the wrong ip:
`podman run -it --network slirp4netns:port_handler=slirp4netns --publish 192.168.178.119:8080:80 alpine nc -v -n  -l -p 80`
This works for me, when I connect I see `192.168.178.119` as remote ip.

**Reply by FGasper** [reply]
@Luap99 The local IP is what I need, actually.

Context: I have 2 httpd instances, one containerized and the other non-containerized. I’m trying to copy the host’s httpd.conf to the container. Ideally I’d like, at least for my demonstration purposes, to keep the IPs in the <VirtualHost> entries the same.

So if httpd on the host listens on 10.2.65.75:80, I’d like the containerized httpd to do the same, with podman just mapping the container’s port 80 to the host’s 8080.

**Reply by Luap99** [reply]
This is not possible the slir4netns port handler just keeps the source ip.

With slirp4netns you cannot chnage the local ip. You could change the subnet with the cidr option but it will always take the .100 from that.

I think https://passt.top/passt/about/#pasta sounds more like what you want then but this is not yet integrated into podman.
